web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -124,6 +127,17 @@ interface UnifiedResourcesProps {
   availableKinds: SharedUnifiedResource['resource']['kind'][];
   setParams(params: UnifiedResourcesQueryParams): void;
   onLabelClick(label: ResourceLabel): void;
+  // bulkActions allow us to pass in any features we want from the parent
+  // that interact with selected items (example: add all selected resources to an access request)
+  // TODO (avatus): pinning should be removed from the internals of UnifiedResources as passed in
+  // from the parents.
+  bulkActions: {
+    // key is an arbitrary name of what the bulk action is, as well
+    // as the key used when mapping our action components
+    key: string;
+    Component: (onClick: () => void) => React.ReactElement;
+    action: (selectedResourceIds: string[]) => void;
```

Comment:
Now that I think of it and your use case, a nicer API would be more like this:
```ts
    action: (
      selectedResources: {
        unifiedResourceId: string;
        resource: SharedUnifiedResource['resource'];
      }[]
    ) => void
```
Instead of mapping the items outside of the component, we can do it inside it and simplify the callsites.